src/waveform/renderers/waveformmark.cpp

```diff
@@ -5,21 +5,28 @@
 
 #include "waveformmark.h"
 
-WaveformMark::WaveformMark(int hotCue)
+WaveformMark::WaveformMark( const QString& group,
+                            const QDomNode& node,
+                            const SkinContext& context,
+                            const WaveformSignalColors& signalColors,
+                            int hotCue)
     : m_iHotCue(hotCue) {
-}
-
-void WaveformMark::reset(int hotCue) {
-    WaveformMark(hotCue).swap(*this);
-}
-
-void WaveformMark::setup(const QString& group, const QDomNode& node,
-                         const SkinContext& context,
-                         const WaveformSignalColors& signalColors) {
     QString item = context.selectString(node, "Control");
     if (!item.isEmpty()) {
```

Comment:
The question here is: If we have a skin with a WavefromMark and no Control. Is this valid? Or can we check this outside and skip the crating of the WaveformMark?